c2corg_ui/static/js/preferences.js

```diff
@@ -0,0 +1,157 @@
+goog.provide('app.PreferencesController');
+goog.provide('app.preferencesDirective');
+
+goog.require('app');
+goog.require('app.Alerts');
+goog.require('app.Api');
+goog.require('app.Authentication');
+goog.require('app.utils');
+
+
+/**
+ * Directive managing the user preferences.
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.preferencesDirective = function() {
+  return {
+    restrict: 'A',
+    scope: true,
+    controller: 'appPreferencesController',
+    controllerAs: 'prefCtrl',
+    bindToController: true
+  };
+};
+
+
+app.module.directive('appPreferences', app.preferencesDirective);
+
+/**
+ * @param {angular.Scope} $scope Scope.
+ * @param {app.Authentication} appAuthentication
+ * @param {app.Alerts} appAlerts
+ * @param {app.Api} appApi Api service.
+ * @param {string} authUrl Base URL of the authentication page.
+ * @constructor
+ * @ngInject
+ * @export
+ */
+app.PreferencesController = function($scope, appAuthentication, appAlerts,
+    appApi, authUrl) {
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  /**
+   * @type {app.Alerts}
+   * @private
+   */
+  this.alerts_ = appAlerts;
+
+  /**
+   * @type {app.Api}
+   * @private
+   */
+  this.api_ = appApi;
+
+  /**
+   * @type {Array.<string>}
+   * @export
+   */
+  this.activities = [];
+
+  /**
+   * @type {Array.<appx.Area>}
+   * @export
+   */
+  this.areas = [];
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.followed_only = false;
+
+  if (appAuthentication.isAuthenticated()) {
+    this.api_.readPreferences().then(function(response) {
+      var data = /** @type {appx.UserPreferences} */ (response['data']);
+      this.activities = data.activities;
+      this.areas = data.areas;
+      this.followed_only = data.followed_only;
+
+      this.scope_.$watch(function() {
+        return this.followed_only;
+      }.bind(this), function(newValue, oldValue) {
+        if (newValue !== oldValue) {
+          this.save_();
+        }
+      }.bind(this));
+    }.bind(this));
+  } else {
+    app.utils.redirectToLogin(authUrl);
+  }
+};
+
+
+/**
+ * @param {string} activity
+ * @export
+ */
+app.PreferencesController.prototype.updateActivities = function(activity) {
+  if (this.activities.indexOf(activity) > -1) {
+    this.activities = this.activities.filter(function(item) {
+      return item !== activity;
+    });
+  } else {
+    this.activities.push(activity);
+  }
+  this.save_();
+};
+
+
+/**
+ * @param {appx.Area} area
+ * @export
+ */
+app.PreferencesController.prototype.addArea = function(area) {
+  var id = area.document_id;
+  for (var i = 0, n = this.areas.length; i < n; i++) {
```

Comment:
Thanks for the tip! But I am not really convinced there's a gain using it because it's as long and a little less readable (because "some" is not very well known to most developers?).

* A:
```
  for (var i = 0, n = this.areas.length; i < n; i++) {
    if (this.areas[i].document_id === id) {
      // selected area is already in the list
      return;
    }
  }
```
* B:
```
  if (this.areas.some(function(a) {
    return a.document_id === id;
  })) {
    // selected area is already in the list
    return;
  }
```
I think I prefer A.